src/main/java/org/junitpioneer/jupiter/DefaultLocale.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * {@code @DefaultLocale} is a JUnit Jupiter extension to change the value
+ * returned by {@link java.util.Locale#getDefault()} for a test execution.
+ *
+ * <p>The {@link java.util.Locale} to set as the default locale can be
+ * configured using only a language, a language and a country or a language, a
+ * country and a variant. If {@link #variant()} is set, but {@link #country()}
+ * is not, a
+ * {@link org.junit.jupiter.api.extension.ExtensionConfigurationException} will
+ * be thrown. After the annotated element has been executed, the default Locale
+ * will be restored to its original value.</p>
+ *
+ * <p>{@code @DefaultLocale} can be used on the method and on the class level.
+ * If a class is annotated, the configured Locale will be the default Locale
+ * for all tests inside that class. Any method level configurations will
+ * override the class level default Locale.</p>
+ *
+ * @since 0.2
+ * @see java.util.Locale#getDefault()
+ */
+@Target({ ElementType.METHOD, ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+@ExtendWith(DefaultLocaleExtension.class)
+public @interface DefaultLocale {
+
+	String language();
```

Comment:
I'd like to have `language()` as alias for `value()`. This is possible with `@AliasFor` from spring framework. Is there something comparable in JUnit?

Reply:
No. And their we won't  add `@AliasFor` support in the near future. There exists a ticket regarding that feature ...

Reply:
I'd prefer `value()`  over `language()` here. But that's debatable.

Reply:
I don't have a strong opinion on this. While `value()` is more concise, using `language()` hat the advantage that it follows the naming of the constructor parameters of java.util.Locale. It's your call ;-)

Reply:
Since there are three `String` attributes in the annotation and the likelihood that one would need to supply more than just the language is rather high, I'd prefer `language` in this particular use case.

Reply:
I prefer `language`, too.

Isn't it common to write the locale in [snake case](https://en.wikipedia.org/wiki/Snake_case)? In that case, `value` could accept a string like `"en_EN_gb"` and take it apart. We don't need that from the get go, though, and if this is not totally harebrained, we should discuss it in a separate issue.

Reply:
Another thing you might consider:

https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html#forLanguageTag-java.lang.String-

`@DefaultLocale(languageTag = "en-US")`

would be more succinct than

`@DefaultLocale(language = "en", country="US")`